userProfile/src/Components/Form.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import { Button, Input, Gapped, Select, Modal } from '@skbkontur/react-ui';
+
+const cities: string[] = ['Екатеринбург', 'Н.Тагил', 'Москва', 'Санкт-Петербург', 'Казань'];
+
+interface FormData {
+    name: string;
+    surname: string;
+    city: string | undefined;
+}
+
+const saveData: FormData = {
+    name: '',
+    surname: '',
+    city: ''
+};
+
+const defaultData: FormData = {
+    name: '',
+    surname: '',
+    city: cities[0]
+};
+
+const Form = () => {
+    const [name, setName] = useState(defaultData.name);
+    const [surname, setSurname] = useState(defaultData.surname);
+    const [city, setCity] = useState(defaultData.city);
+
+    const [person, setPerson] = useState({
+        name,
+        surname,
+        city
+    });
+
+    const [opened, setOpened] = useState(false);
+
+    const handlerField = (field: keyof FormData, value: string) => {
+        switch (field) {
+            case 'name':
+                setName(value);
+                break;
+            case 'surname':
+                setSurname(value);
+                break;
+            case 'city':
+                setCity(value);
+                break;
+            default:
+                throw new Error();
+        }
+    };
+
+    const submitForm = () => {
+        setPerson({ name: name, surname: surname, city: city });
```

Comment:
Можно не писать такое строгое присвоение, если поле в объекте совпадает с названием переменной, в которой храниться нужное значение
```suggestion
        setPerson({ name, surname, city });
```